src/bootstrap/src/utils/exec.rs

```diff
@@ -44,17 +44,61 @@ impl<'a> BootstrapCommand<'a> {
         Self { failure_behavior: BehaviorOnFailure::Ignore, ..self }
     }
 
+    /// Do not print the output of the command, unless it fails.
+    pub fn quiet(self) -> Self {
+        self.output_mode(OutputMode::OnlyOnFailure)
+    }
+
     pub fn output_mode(self, output_mode: OutputMode) -> Self {
-        Self { output_mode, ..self }
+        Self { output_mode: Some(output_mode), ..self }
     }
 }
 
 impl<'a> From<&'a mut Command> for BootstrapCommand<'a> {
     fn from(command: &'a mut Command) -> Self {
-        Self {
-            command,
-            failure_behavior: BehaviorOnFailure::Exit,
-            output_mode: OutputMode::PrintAll,
-        }
+        Self { command, failure_behavior: BehaviorOnFailure::Exit, output_mode: None }
+    }
+}
+
+/// Represents the output of an executed process.
+#[allow(unused)]
+#[derive(Default)]
+pub struct CommandOutput {
+    status: ExitStatus,
+    stdout: Vec<u8>,
+    stderr: Vec<u8>,
+}
```

Comment:
What about wrapping the actual type?

```suggestion
pub struct CommandOutput(std::process::Output)
```

Reply:
I did that originally, but I thought that you cannot create `Output` manually from outside stdlib (which we need e.g. for dry runs). But it turns out that you can! So I changed it as you have proposed, thanks.